### PR TITLE
[WTF-1088] Add documentation for PW association property [9.13]

### PR DESCRIPTION
**Pull request by mlaponder**

### CODE REVIEW

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md [hunk 1]
```diff
@@ -137,7 +137,49 @@ There is a way to use more the convenient `displayValue`  and `setTextValue` whi
...
+}
+
+if (association.type === "Reference") {
+	return textTemplate.get(association.value);
+} else {
+	return association.value.map((objectItem) => textTemplate.get(objectItem)).join(",");
+}
+```
+
+`status` is similar to the one exposed for `DynamicValue`. It indicates if the value's loading has finished and if loading was successful. Similarly to `DynamicValue`, `EditableValue` keeps returning the previous `value` when `status` changes from `Available` to `Loading` to help a widget avoid flickering.
```
Comment (tamelbrun) [inline content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: I suspect that this was copied from `EditableValue`, but it should be `ModifiableValue` here ;)

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md [hunk 2]
```diff
@@ -137,7 +137,49 @@ There is a way to use more the convenient `displayValue`  and `setTextValue` whi
 
 The optional field `universe` is used to indicate the set of all possible values that can be passed to a `setValue` if a set is limited. Currently, `universe` is provided only when the edited attribute is of the Boolean or enumeration [types](/refguide/attributes/#type).
 
-### 4.4 IconValue {#icon-value}
+### 4.4 ModifiableValue {#modifiable-value}
+
+ModifiableValue is used to represent values that can be changed by a pluggable widget client component and is passed only to [association properties](/apidocs-mxsdk/apidocs/pluggable-widgets-property-types/#association). It is defined as follows:
```
Comment (tamelbrun) [inline content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: Should `ModifiableValue` (and similarly, `EditableValue` in the first sentence of that section) be formatted as code here?

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md [hunk 3]
```diff
@@ -137,7 +137,49 @@ There is a way to use more the convenient `displayValue`  and `setTextValue` whi
...
+	readonly readOnly: boolean;
+    
+	readonly value: Option<T>;
+	readonly setValue: (value: Option<T>) => void;
+	readonly validation: Option<string>;
+	readonly setValidator: (validator?: (value: Option<T>) => Option<string>) => void;
+}
+```
+
+The type the component receives for the association property depends on the allowed association types. If only references are allowed the component receives a `ReferenceValue` which is defined as `ModifiableValue<ObjectItem> & { type: "Reference" };`, if only reference sets are allowed the client gets a `ReferenceSetValue` which is defined as `ModifiableValue<ObjectItem[]> & { type: "ReferenceSet" };`.
```
Comment (tamelbrun) [inline content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: I would say "The type received by the component" instead, but maybe @ConnorLand would already pick up on this when dotting all the i's and crossing all the t's :P.

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md [hunk 4]
```diff
@@ -137,7 +137,49 @@ There is a way to use more the convenient `displayValue`  and `setTextValue` whi
 
 The optional field `universe` is used to indicate the set of all possible values that can be passed to a `setValue` if a set is limited. Currently, `universe` is provided only when the edited attribute is of the Boolean or enumeration [types](/refguide/attributes/#type).
 
-### 4.4 IconValue {#icon-value}
+### 4.4 ModifiableValue {#modifiable-value}
```
Comment (tamelbrun) [inline content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md:140]: @ConnorLand, do you see any problems with changing the numbering of existing sections? (I do think the order of the sections makes more sense this way.)

**Review by tamelbrun — CHANGES_REQUESTED** [review]

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 1 above)
Comment (mlaponder) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: Ah, sloppy pasted indeed 🤦 , fixed

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 2 above)
Comment (mlaponder) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: I think it should, also updated some other types as well

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 3 above)
Comment (mlaponder) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: I agree, fixed

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 4 above)
Comment (ConnorLand) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md:140]: I like the new way. Changing the numbers should not be a problem.

Comment (tamelbrun) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md:140]: Awesome, then we'll keep it like this :).

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 3 above)
Comment (tamelbrun) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: ✨

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 1 above)
Comment (tamelbrun) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: ✨

File: content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md (hunk 2 above)
Comment (tamelbrun) [reply content/en/docs/apidocs-mxsdk/apidocs/pluggable-widgets/pluggable-widgets-client-apis/_index.md]: I noticed, noice! ✨

**Review by tamelbrun — APPROVED** [review]

**Review by ConnorLand — APPROVED** [review]

**Pull request merged**